staging/src/k8s.io/kubectl/pkg/kuberc/kuberc.go

```diff
@@ -0,0 +1,336 @@
+	for key, val := range aliasArgsMap {
+		p.aliases[key] = struct{}{}
+		rootCmd.AddCommand(val.command)
+	}
+
+	var aliasName string // first "non-flag" arguments
+	for _, arg := range args[1:] {
+		if !strings.HasPrefix(arg, "-") {
+			aliasName = arg
+			break
+		}
+	}
+
+	foundAliasCmd, _, err := rootCmd.Find([]string{aliasName})
+	if err != nil {
+		return args, nil
+	}
+
+	aliasArgs, ok := aliasArgsMap[aliasName]
+	if !ok {
+		return args, nil
+	}
+
+	// This function triggers merging the persistent flags in the parent commands.
+	_ = foundAliasCmd.InheritedFlags()
+
+	for _, fl := range aliasArgs.flags {
+		existingFlag := foundAliasCmd.Flag(fl.Name)
+		if existingFlag == nil {
+			return args, fmt.Errorf("invalid alias flag %s in alias %s", fl.Name, args[0])
+		}
+		if searchInArgs(existingFlag.Name, existingFlag.Shorthand, args) {
+			// Don't modify the value implicitly, if it is passed in args explicitly
+			continue
+		}
+		err = foundAliasCmd.Flags().Set(fl.Name, fl.Default)
+		if err != nil {
+			return args, fmt.Errorf("could not apply value %s to flag %s in alias %s err: %w", fl.Default, fl.Name, args[0], err)
+		}
+	}
+
+	// all args defined in kuberc should be appended to actual args.
+	args = append(args, aliasArgs.args...)
+	// Cobra (command.go#L1078) appends only root command's args into the actual args and ignores the others.
+	// We are appending the additional args defined in kuberc in here and
+	// expect that it will be passed along to the actual command.
+	rootCmd.SetArgs(args[1:])
+	return args, nil
+}
+
+// DefaultGetPreferences returns KubeRCConfiguration.
+// If users sets kuberc file explicitly in --kuberc flag, it has the highest
+// priority. If not specified, it looks for in KUBERC environment variable.
+// If KUBERC is also not set, it falls back to default .kuberc file at the same location
+// where kubeconfig's defaults are residing in.
+func DefaultGetPreferences(kuberc string) (*config.Preferences, error) {
+	kubeRCFile := RecommendedKubeRCFile
+	explicitly := false
+	if kuberc != "" {
+		kubeRCFile = kuberc
+		explicitly = true
+	}
+
+	if kubeRCFile == "" && os.Getenv("KUBERC") != "" {
+		kubeRCFile = os.Getenv("KUBERC")
+		explicitly = true
+	}
+
+	kubeRCBytes, err := os.ReadFile(kubeRCFile)
+	if err != nil {
+		if os.IsNotExist(err) && !explicitly {
+			return nil, nil
+		}
+		return nil, err
+	}
+	pref := &config.Preferences{}
+	_, gvk, err := scheme.Codecs.UniversalDecoder(v1alpha1.SchemeGroupVersion).Decode(kubeRCBytes, nil, pref)
+	if err != nil {
+		return nil, fmt.Errorf("could not be decoded gvk %s, err: %w", gvk, err)
+	}
+	return pref, nil
+}
+
+// Normally, we should extract this value directly from kuberc flag.
+// However, flag values are set during the command execution and
+// we are in very early stages to prepare commands prior to execute them.
+// Besides, we only need kuberc flag value in this stage.
+func getExplicitKuberc(args []string) string {
```

Comment:
hmm, this seems buggy if it is extracting flags without understanding whether they are positional args or flags...

it will accidentally think a flag value or positional arg is this flag, e.g.

```
kubectl exec pod/mypod -- kubectl ... --kuberc=...
```

Reply:
Nice catch. It is not correct continuing the iteration once we encounter `--`. Updated according to this and added unit tests for the feature parity.